core/templates/components/forms/custom-forms-directives/image-uploader.directive.ts

```diff
@@ -51,18 +53,48 @@ angular.module('oppia').directive('imageUploader', [
             return 'This file is not recognized as an image.';
           }
 
-          if (!file.type.match('image/jpeg') &&
-            !file.type.match('image/gif') &&
-            !file.type.match('image/jpg') &&
-            !file.type.match('image/png') &&
-            !file.type.match('image/svg\\+xml')) {
+          var imageTypeMapping = {
+            jpeg: {
+              format: 'image/jpeg',
+              condition: [/jp(e?)g$/, /\.jp(e?)g$/],
+            },
+            jpg: {
+              format: 'image/jpg',
+              condition: [/jp(e?)g$/, /\.jp(e?)g$/],
+            },
+            gif: {
+              format: 'image/gif',
+              condition: [/gif$/, /\.gif$/],
+            },
+            png: {
+              format: 'image/png',
+              condition: [/png$/, /\.png$/],
+            },
+            svg: {
+              format: 'image/svg\\+xml',
+              condition: [/svg\+xml$/, /\.svg$/],
+            }
+          };
+
+          var imageFormatCheck = true;
+          var imageExtensionCheck = false;
+          var allowedImageType = JSON.parse(scope.allowedImageType);
+          for (var i = 0; i < allowedImageType.length; i++) {
+            var fileType = allowedImageType[i];
+            imageFormatCheck = (
+              imageFormatCheck &&
+              !file.type.match(imageTypeMapping[fileType].format));
+            imageExtensionCheck = (
+              imageExtensionCheck ||
+              (file.type.match(imageTypeMapping[fileType].condition[0]) &&
```

Comment:
This is not good design because it always assumes "condition" has 2 elements, which is not enforced anywhere. Drop the array and make it a single regex string. Or have a list of allowed regexes and iterate through them. But don't refer to [0] and [1].

Reply:
These are two different regex each for the filetype and extension so I have split the array into two different elements. Done.